src/serverExplorer.ts

```diff
@@ -284,7 +284,12 @@ export class ServerExplorer implements TreeDataProvider<RSPState | ServerStateNo
         // Windows -> if both options (canSelectFiles and canSelectFolders) are true, fs only shows folders
         // Linux(fedora) -> if both options are true, fs shows both files and folders but files are unselectable
         // Mac OS -> if both options are true, it works correctly
+        var d = workspace.workspaceFile != undefined ? workspace.workspaceFile : 
+                workspace.workspaceFolders == undefined ? undefined : 
+                workspace.workspaceFolders.length <= 0 ? undefined : 
+                workspace.workspaceFolders[0].uri;
```

Comment:
Some considerations

1) better to use `const` instead of `var`
2) I would give a more significant name to the variable
3) always prefer the triple equals (=== / !==) which makes a stricter control otherwise you can write the same with !workspace.workspaceFile and write less code
4) Not quite sure the order is correct. 
Let's suppose you have added a folder in your workspace, then you open a single external file in VSCode. At this point the `add deployment` dialog will always be opened in the folder of this file because this is the first check made. But i guess it would be more useful to still open the folder added firstly.

Other thing to think about is if the user has more than one folder added in the workspace. Let's say i add `project A` to the workspace, then `project B`. The `add deployment` dialog will always open to the first folder `workspace.workspaceFolders[0].uri`. But what if i'm working on project B right now? Maybe you could think of taking the path from the file currently selected by using `window.activeTextEditor.document.uri`.

Reply:
1) const instead of var +1
2) naming var +1
3) tripple-equals +1
4) Looking into it